src/parser/jobs/brd/PitchPerfect.js

```diff
@@ -0,0 +1,253 @@
+		// Relying on the normaliser from this for the hit type fields
+		'hitType', // eslint-disable-line xivanalysis/no-unused-dependencies
+	]
+
+	// Represents a map of IDs and statuses for each enemy in this parse
+	_enemies = {}
+
+	// Represents the player statuses
+	_player = {
+		statuses: {},
+		hasStatus(statusId) { return this.statuses[statusId] || false },
+	}
+
+	normalise(events) {
+
+		const damageInstances = []
+		const ppEvents = []
+
+		for (const event of events) {
+			// Registers buffs/debuffs statuses on the respective entity (either player or enemies)
+			if (event.type.startsWith('apply') || event.type.startsWith('remove')) {
+				if (event.targetID === this.combatants.selected.id && event.ability) {
+					this._player.statuses[event.ability.guid] = event.type.startsWith('apply')
+				} else if (!event.targetIsFriendly) {
+					const enemy = this._getEnemy(event.targetID)
+					enemy.statuses[event.ability.guid] = event.type.startsWith('apply')
+				}
+
+			// For every damage event that:
+			// - comes from the player
+			// - is not a dot tick
+			} else if (
+				event.type === 'damage'
+				&& event.sourceID === this.combatants.selected.id
+				&& event.ability
+				&& !event.tick
+			) {
+				// Fixing the multiplier
+				let fixedMultiplier = event.debugMultiplier
+				if (
+					event.ability.guid !== ACTIONS.THE_WANDERERS_MINUET.id
+					&& event.ability.guid !== ACTIONS.MAGES_BALLAD.id
+					&& event.ability.guid !== ACTIONS.ARMYS_PAEON.id
+				) {
+					// Band-aid fix for disembowel (why, oh, why)
+					if (this._getEnemy(event.targetID).hasStatus(STATUSES.PIERCING_RESISTANCE_DOWN.id)) {
+						fixedMultiplier = Math.trunc((fixedMultiplier + DISEMBOWEL_STRENGTH) * 100) / 100
+					}
+					// AND ALSO FOR RANGED TRAIT, BECAUSE APPARENTLY IT'S PHYSICAL DAMAGE ONLY REEEEEEEEEE
+					fixedMultiplier = Math.trunc((fixedMultiplier + TRAIT_STRENGTH) * 100) / 100
+				}
+
+				// If it's a Pitch Perfect damage
+				if (event.ability.guid === ACTIONS.PITCH_PERFECT.id) {
+
+					ppEvents.push({event: event, rawDamage: event.amount / fixedMultiplier})
+
+				// Otherwise, if it doesn't have a conditional potency (Sidewinder and Pitch Perfect), it will be used to calculate 'K'
+				} else if (event.ability.guid !== ACTIONS.SIDEWINDER.id) {
+
+					// ...let's not count Spears for now
+					if (!this._player.hasStatus(STATUSES.THE_SPEAR.id)) {
+
+						const critTier = this._parseCritBuffs(event)
+
+						if (!damageInstances[critTier]) {
+							damageInstances[critTier] = []
+						}
+
+						damageInstances[critTier].push({event: event, rawDamage: event.amount / fixedMultiplier})
+					}
+				}
+			}
+		}
+
+		// We use the damage events to determine 'K'
+		// tl;dr: 'K' is an approximation to damage to potency ratio, ignoring the natural 5% spread because we don't need this kind of precision
+		const k = this._getK(damageInstances)
+
+		// We now use 'K' to guesstimate PP potency:
+		for (const pp of ppEvents) {
```

Comment:
Not going to block on it for now, but I question if this second loop needs to executed in the normalisation step. I can imagine a lot of the logic in this module could be generalised, and the PP-specific handling moved into the main loop.

Reply:
I'll look into generalizing it if there's enough reason to. Right now, I can only assure it has enough precision for specifically PP's case. PP's potencies are spread enough that a lot of approximations don't change the end result.

I'm not sure if that would be the case in, say, figuring out a weaponskill potency to see if a positional was missed. I doubt it offers enough precision to differentiate between a 140 potency and a 180 potency True Strike.